devolutions-gateway/src/api/monitoring.rs

```diff
@@ -0,0 +1,224 @@
+        (status = 400, description = "Bad request"),
+        (status = 401, description = "Invalid or missing authorization token"),
+        (status = 403, description = "Insufficient permissions"),
+        (status = 500, description = "Unexpected server error while starting monitors"),
+    ),
+))]
+async fn handle_set_monitoring_config(
+    extract::State(DgwState { monitoring_state, .. }): extract::State<DgwState>,
+    Json(config): Json<MonitorsConfig>,
+) -> Result<Json<SetConfigResponse>, HttpError> {
+    let (processed_config, probe_type_errors) = config.lossy_into();
+
+    network_monitor::set_config(processed_config, monitoring_state)
+        .await
+        .map(|_| Json(SetConfigResponse::new(probe_type_errors)))
+        .map_err(
+            HttpError::internal()
+                .with_msg("Failed to set up network monitoring")
+                .err(),
+        )
+}
+
+/// Monitors store their results in a temporary log, which is returned here.
+/// Once the log is downloaded, gateway purges it from memory.
+#[cfg_attr(feature = "openapi", utoipa::path(
+    post,
+    operation_id = "DrainMonitoringLog",
+    tag = "NetworkMonitoring",
+    path = "/jet/net/monitor/log/drain",
+    responses(
+        (status = 200, description = "Log was flushed and returned in the response body", body = MonitoringLogResponse),
+        (status = 400, description = "Bad request"),
+        (status = 401, description = "Invalid or missing authorization token"),
+        (status = 403, description = "Insufficient permissions"),
+        (status = 500, description = "Unexpected server error"),
+    ),
+))]
+async fn handle_drain_log(
+    extract::State(DgwState { monitoring_state, .. }): extract::State<DgwState>,
+) -> Json<MonitoringLogResponse> {
+    Json(MonitoringLogResponse {
+        entries: network_monitor::drain_log(monitoring_state)
+            .into_iter()
+            .map(MonitorResult::from)
+            .collect(),
+    })
+}
+
+#[cfg_attr(feature = "openapi", derive(utoipa::ToSchema))]
+#[derive(Debug, Serialize, Deserialize)]
+pub struct MonitorsConfig {
+    monitors: Vec<MonitorDefinition>,
+}
+
+impl MonitorsConfig {
+    fn lossy_into(self) -> (network_monitor::MonitorsConfig, Vec<MonitorDefinitionProbeTypeError>) {
+        let (monitors, errors): (
+            Vec<network_monitor::MonitorDefinition>,
+            Vec<MonitorDefinitionProbeTypeError>,
+        ) = self.monitors.into_iter().map(MonitorDefinition::try_into).fold(
+            (
+                Vec::<network_monitor::MonitorDefinition>::new(),
+                Vec::<MonitorDefinitionProbeTypeError>::new(),
+            ),
+            |mut partitions, conversion_result| {
+                match conversion_result {
+                    Ok(value) => partitions.0.push(value),
+                    Err(error) => partitions.1.push(error),
+                };
+
+                return partitions;
+            },
+        );
```

Comment:
style: Probably linted by clippy

```suggestion
        let config = network_monitor::MonitorsConfig { monitors };
```